docs/user-guide/02-conventional-features.md

```diff
@@ -792,3 +792,394 @@ In this example, `globalUniform` is appended to the global root signature / pipe
 Compute entry points lack "local root signatures" in D3D12, and likewise Vulkan has no concept of "local" vs "global" compute pipeline layouts, so `localUniform1` is "pushed" to the stack of reserved global uniform parameters for use in `computeMain1`. 
 Leaving that entry point scope "pops" that global uniform parameter such that `localUniform2` can reuse the same binding location for `computeMain2`.
 However, local uniforms for ray tracing shaders map to the corresponding "local" hit records in the shader binding table, and so no "push" or "pop" to the global root signature / pipeline layouts occurs for these parameters. 
+
+Auto-Generated Constructors
+----------
+
+### Auto-Generated Constructors - Struct
+
+Slang has the following rules:
+1. Auto-generate a `__init()` if not already defined
+> Assume
+```c#
+struct DontGenerateCtor
+{
+    int a;
+    int b = 5;
+
+    __init()
```

Comment:
Add a comment here:
// because the user has explicitly defined a constructor, 
// slang will not synthesize more constructors.